src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -454,36 +458,25 @@ public bool ContainsDirective(int rawKind)
                 while (stack.Count > 0)
                 {
                     var current = stack.Pop();
-                    if (current is null)
-                        continue;
 
                     // Don't bother looking further down this portion of the tree if it clearly doesn't contain directives.
-                    if (!current.ContainsDirectives)
+                    if (current is not { ContainsDirectives: true })
```

Comment:
Doesn't feel like this an idiomatic use of patterns. Consider putting it back.

Reply:
I def disagree :)  We're using this on the IDE a lot as the synonym for `if (x is null || !x.Blah) continue;`  def seems much cleaner and easier (imo at least).

Reply:
I would agree, if the null test was relevant. It's not.

Reply:
It's necessary here. Child slots can be null, so this stack can contain null items :-)